File: 334.increasing-triplet-subsequence.py
#
# @lc app=leetcode id=334 lang=python3
#
# [334] Increasing Triplet Subsequence
#

# Time:  O(n)
# Space: O(1)

from typing import List


# @lc code=start
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first, second = float("inf"), float("inf")
        for num in nums:
            if first >= num:
                first = num
            elif second >= num:
                second = num
            else:
                return True
        return False


# @lc code=end

File: 334.increasing-triplet-subsequence.js
/*
 * @lc app=leetcode id=334 lang=javascript
 *
 * [334] Increasing Triplet Subsequence
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function (nums) {
  let first = Infinity;
  let second = Infinity;
  for (const num of nums) {
    if (first >= num) {
      first = num;
    } else if (second >= num) {
      second = num;
    } else {
      return true;
    }
  }
  return false;
};
// @lc code=end
